Add troubleshooting note for high custom metric billings when enabling openmetrics (#10545)
DataDog/integrations-core: istio/README.md

## Troubleshooting

> ### Using Openmetrics Integration

Reviewer: ```suggestion
### Duplicate custom metric collection while also using Openmetrics integration
```

> If Istio proxy sidecar injection is enabled, monitoring other Prometheus metrics via the [Openmetrics integration][20] can result in high custom metrics usage and duplicated metric collection.

Reviewer 2: ```suggestion
If you enable the Istio proxy sidecar injection, monitoring other Prometheus metrics with the [Openmetrics integration][20] may result in high custom metrics usage and duplicated metric collection.
```

> To ensure that your Openmetrics configuration is not redundantly collecting metrics, either:

Reviewer 2: ```suggestion
To ensure that your Openmetrics configuration does not redundantly collect metrics, either:
```

> 1. Use specific metric matching in the `metrics` configuration option

Reviewer 2: ```suggestion
1. Use specific metric matching in the `metrics` configuration option, or
```

> 2. If you use `*` value for `metrics`, then at minimum, consider including the following optional options to exclude metrics already collected by istio and Envoy inegrations.

Reviewer 3: ```suggestion
2. If you use `*` value for `metrics`, then at minimum, consider including the following optional options to exclude metrics already collected by the Istio and Envoy integrations.
```

> 2. If you use `*` value for `metrics`, then at minimum, consider including the following optional options to exclude metrics already collected by istio and Envoy inegrations.

Reviewer: ```suggestion
2. If using the wildcard `*` value for `metrics`, then consider including the following optional options in the Istio or Envoy `conf.yaml` to exclude metrics already collected.
```

Author: the second half of your suggestion is incorrect. The recommendations below are for the openmetrics configuration... you're not suppose to exclude metrics from Istio or Envoy conf.yaml.

Reviewer: Good catch, updated comment to be openmetrics

> 2. If you use `*` value for `metrics`, then at minimum, consider including the following optional options to exclude metrics already collected by the Istio and Envoy integrations.

Reviewer 2: ```suggestion
2. If you use `*` value for `metrics`, consider using the following options to exclude metrics already collected by the Istio and Envoy integrations.
```

    ##
    ##      a. If a value is a string, then it represents the sent metric name. For example:
    ##
    ##           metrics:
    ##           - <EXPOSED_METRIC_1>: <SENT_METRIC_1>
    ##           - <EXPOSED_METRIC_2>: <SENT_METRIC_2>
    ##      b. If a value is a mapping, then it must have a `name` and/or `type` key.
    ##         The `name` represents the sent metric name, and the `type` represents how
    ##         the metric should be handled, overriding any type information the endpoint
    ##         may provide. For example:
    ##
    ##           metrics:
    ##           - <EXPOSED_METRIC_1>:
    ##               name: <SENT_METRIC_1>
    ##               type: <METRIC_TYPE_1>
    ##           - <EXPOSED_METRIC_2>:
    ##               name: <SENT_METRIC_2>
    ##               type: <METRIC_TYPE_2>
    ##
    ##         The supported native types are `gauge`, `counter`, `histogram`, and `summary`.
    ##
    ## Regular expressions may be used to match the exposed metric names, for example:
    ##
    ##   metrics:
    ##   - ^network_(ingress|egress)_.+
    ##   - .+:
    ##       type: gauge

>     #

Reviewer: I think we can leave out the comments, since they can find it in `conf.yaml.example`. WDYT?

Author: hmm, my concern with that is it will be confusing if they erroneously assume this to be for Istio conf.yaml.example. I wanted users to know exactly what they're enabling.

Note: you must upgrade to at minimum Agent `7.31.0` and Python 3. See the [Configuration](#configuration) section on enabling Openmetrics V2.

### Using the generic Openmetrics Integration in an Istio deployment

If Istio proxy sidecar injection is enabled, monitoring other Prometheus metrics via the [Openmetrics integration][21] with the same metrics endpoint as `istio_mesh_endpoint` can result in high custom metrics usage and duplicated metric collection.

To ensure that your Openmetrics configuration does not redundantly collect metrics, either:

1. Use specific metric matching in the `metrics` configuration option, or
2. If using the wildcard `*` value for `metrics`, consider using the following Openmetrics integration options to exclude metrics already supported by the Istio and Envoy integrations.

#### Openmetrics V2 configuration with generic metric collection

> Be sure to exclude Istio and Envoy metrics from your configuration to avoid high custom metrics billing. Use `exclude_metrics` if configuring the Openmetrics V2 (`openmetrics_endpoint` enabled).

Reviewer: ```suggestion
Be sure to exclude Istio and Envoy metrics from your configuration to avoid high custom metrics billing. Use `exclude_metrics` if using the Openmetrics V2 configuration (`openmetrics_endpoint` enabled).
```

 
```yaml
## Every instance is scheduled independent of the others.
#
instances:
  - openmetrics_endpoint: <OPENMETRICS_ENDPOINT>
    metrics: [*]
    exclude_metrics:
      - istio_*
      - envoy_*

```

#### Openmetrics V1 configuration (Legacy) with generic metric collection

> Be sure to exclude Istio and Envoy metrics from your configuration to avoid high custom metrics billing. Use `ignore_metrics` if configuring the Openmetrics V2 (`openmetrics_endpoint` enabled).

Reviewer: ```suggestion
Be sure to exclude Istio and Envoy metrics from your configuration to avoid high custom metrics billing. Use `ignore_metrics` if using the Openmetrics V1 configuration (`prometheus_url` enabled).
```
